skiko/src/jvmMain/cpp/common/Shader.cc

```diff
@@ -141,4 +144,28 @@ extern "C" JNIEXPORT jlong JNICALL Java_org_jetbrains_skia_Shader__1nMakeBlend(J
     SkBlendMode blendMode = static_cast<SkBlendMode>(blendModeInt);
     SkShader* ptr = SkShaders::Blend(blendMode, sk_ref_sp<SkShader>(dst), sk_ref_sp<SkShader>(src)).release();
     return reinterpret_cast<jlong>(ptr);
-}
\ No newline at end of file
+}
+
+extern "C" JNIEXPORT jlong JNICALL Java_org_jetbrains_skia_Shader__1nMakeFractalNoise
+  (JNIEnv* env, jclass jclass, jfloat baseFrequencyX, jfloat baseFrequencyY, jint numOctaves, jfloat seed, jintArray tilesArray) {
+    int len = env->GetArrayLength(tilesArray);
+    std::vector<SkISize> tiles(len / 2);
+    jint* arr = env->GetIntArrayElements(tilesArray, 0);
+    for (int i = 0; i < len; i += 2)
+        tiles[i / 2] = {arr[i], arr[i+1]};
+    env->ReleaseIntArrayElements(tilesArray, arr, 0);
+    SkShader* ptr = SkPerlinNoiseShader::MakeFractalNoise(baseFrequencyX, baseFrequencyY, numOctaves, seed, nullptr).release();
```

Comment:
Should be `tiles.data()` instead of `nullptr` here. Not sure how I can amend this thing

Reply:
It's in now